linker/shared/src/main/scala/org/scalajs/linker/frontend/BaseLinker.scala

```diff
@@ -189,9 +190,19 @@ private[frontend] object BaseLinker {
 
     val allMethods = methods ++ syntheticMethodDefs
 
-    val kind =
-      if (classInfo.isModuleAccessed) classDef.kind
-      else classDef.kind.withoutModuleAccessor
+    val kind = {
+      import ClassKind._
+      classDef.kind match {
+        case kind if kind.isJSType && !classInfo.isInstantiated => AbstractJSType
+        case ModuleClass if !classInfo.isModuleAccessed         => Class
+        case JSModuleClass if !classInfo.isModuleAccessed       => JSClass
+        case kind                                               => kind
+      }
+    }
+
+    val jsNativeLoadSpec =
+      if (kind == ClassKind.AbstractJSType) None
```

Comment:
IMO repeating the eliminating condition here is clearer (isInstantiated / isAnySubclassInstantiated).